src/core/tranche/operator.sol

```diff
@@ -0,0 +1,129 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+
+contract ReserveLike{
+    function balance() public view returns (uint);
+    function sliceOf(address) public view returns (uint);
+    function redeem(address, uint, uint) public;
+    function supply(address, uint, uint) public;
+    function repay(address, uint) public;
+    function borrow(address, uint) public;
+}
+
+contract QuantLike {
+   uint public debt; 
+   uint public iBorrow;
+   function file(bytes32, uint) public;
+   function updateDebt(int) public;
+   function UpdateiBorrow(uint, uint) public;
+   function getSpeed() public returns(uint);
+}
+
+contract SlicerLike {
+   function calcSlice(uint) public returns(uint);
+   function calcPayout(uint) public returns(uint);
+   function updateISupply(uint, uint, uint) public;
+}
+
+// Operator
+// Manages the reserve. Triggers iSupply & iBorrow calculations.
+contract Operator is DSNote {
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) public auth note { wards[usr] = 1; }
+    function deny(address usr) public auth note { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+
+    // --- Data ---
+    ReserveLike public reserve;
+    QuantLike public quant;
+    SlicerLike public slicer;
+
+    bool public supplyActive;
+    bool public redeemActive;
+
+    constructor( address reserve_, address quant_, address slicer_) public {
+        wards[msg.sender] = 1;
+        quant = QuantLike(quant_);
+        slicer = SlicerLike(slicer_);
+        reserve = ReserveLike(reserve_);
+        supplyActive = true;
+        redeemActive = true;
+    }
+
+    function file(bytes32 what, bool data) public note auth {
+        if (what == "supply") { supplyActive = data; }
+        else if (what == "redeem") { redeemActive = data; }
+    }
+    
+    function balance() public returns (uint) {
+        return reserve.balance();
+    }
+
+    function supply(address usr, uint currencyAmount) public note auth {
+        require (supplyActive);
+        uint tokenAmount = slicer.calcSlice(currencyAmount);
+        reserve.supply(usr, tokenAmount, currencyAmount);
+        updateSupplyRate();
+    }
+
+    function redeem(address usr, uint tokenAmount) public note auth {
+        require (redeemActive);
+        uint slice = reserve.sliceOf(usr); 
```

Comment:
Why not balance?
```suggestion
        uint balance = reserve.balanceOf(usr); 
```

Reply:
because balanceOf is for currency and sliceOf for the tokens 🙈